zerver/views/messages.py

```diff
@@ -153,9 +153,7 @@ def _pg_re_escape(self, pattern):
         s = list(pattern)
         for i, c in enumerate(s):
             if c not in self._alphanum:
-                if c == '\000':
```

Comment:
I don't find any way to pass the above test cases because we need to remove the NUL char from stream name.

Banning of null characters from stream names sound good to me. Then, we can remove the above test cases and write a fresh test case of stream name validation.